src/DropKit.ts

```diff
@@ -138,29 +160,74 @@ export default class DropKit {
   }
 
   async mint(quantity: number): Promise<void> {
-    if (!this.contract) {
-      throw new Error('Initialization failed')
+    try {
+      // safety check
+      quantity = Number(Math.min(quantity, await this.maxPerMint()))
+
+      const presaleActive = await this.presaleActive()
+      const saleActive = await this.saleActive()
+
+      const tokensCount = await this.walletTokensCount()
+      const maxPerWallet = await this.maxPerWallet()
+
+      if (tokensCount >= maxPerWallet) {
+        throw new Error(
+          `You can't mint more than ${maxPerWallet} tokens on your wallet`
+        )
+      }
+
+      if (!saleActive && !presaleActive) {
+        throw new Error('Collection is not active')
+      }
+
+      const price = await this.price()
+      const amount = ethers.utils.parseEther(price.toString()).mul(quantity)
```

Comment:
`price` should be a Big Number, we can probably just do this: `price.mul(quantity)`

Reply:
Added better typings support, now `_mint` and `_presaleMint` returns `Promise<ContractReceipt>`
 see: https://github.com/niftykit-inc/dropkit.js/pull/6/commits/7215b584295fb3428cd1e99daf45fbc0e3afddfb

Reply:
Changed it, now `price` simply returns a `BigNumber`

```typescript
async price(): Promise<BigNumber> {
    const dropPrice: BigNumber =
      this.version <= 3
        ? await this.contract?._price()
        : await this.contract?.price()

    return dropPrice
  }
```

and now there is no longer "format" and "parse" ethers back and forth:

```typescript
const amount = price.mul(quantity)
```

It's cleaner now!